public/js/icinga/behavior/navigation.js

```diff
@@ -282,63 +314,96 @@
+        const $target = $(this);
+
+        if (! _this.coordinates.includes(undefined) && d3.polygonContains(_this.coordinates, [e.clientX, e.clientY])) {
+            return;
+        }
 
-        if (! $flyout.length) {
+        if (! $target[0].matches(':has(.nav-level-2)')) {
             $layout.removeClass('menu-hovered');
             $target.siblings().not($target).removeClass('hover');
             return;
         }
 
-        var delay = 300;
+        $layout.addClass('menu-hovered');
+        _this.coordinates[0] = [e.clientX, e.clientY];
+        _this.showFlyoutMenu($target, _this.coordinates[0]);
+    }
 
-        if ($layout.hasClass('menu-hovered')) {
-            delay = 0;
+    /**
+     * Captures the mouse move events within the navigation item
+     * and show the flyout if needed.
+     *
+     * @param e
+     */
+    Navigation.prototype.onMouseMove = function(e) {
+        const _this = e.data.self;
+        clearTimeout(_this.flyoutTimer);
+
+        const $target = $(this);
+        _this.coordinates[0] = [e.clientX, e.clientY];
+        if (! _this.coordinates.includes(undefined)) {
+            _this.triangle.attr("points", _this.coordinates.map(p => p.join(",")).join(" "));
         }
 
-        setTimeout(function() {
-            try {
-                if (! $target.is(':hover')) {
-                    return;
-                }
-            } catch(e) { /* Bypass because if IE8 */ }
+        if ($target[0].matches(':has(.nav-level-2)')) {
+            _this.flyoutTimer = setTimeout(function() {
+                _this.showFlyoutMenu($target);
+            }, 200);
+        }
+    };
 
-            $layout.addClass('menu-hovered');
-            $target.siblings().not($target).removeClass('hover');
-            $target.addClass('hover');
-
-            const targetRect = $target[0].getBoundingClientRect();
-            const flyoutRect = $flyout[0].getBoundingClientRect();
-
-            const css = { "--caretY": "" };
-            if (targetRect.top + flyoutRect.height > window.innerHeight) {
-                css.top = targetRect.bottom - flyoutRect.height;
-                if (css.top < 10) {
-                    css.top = 10;
-                    // Not sure why -2, but it aligns the caret perfectly with the menu item
-                    css["--caretY"] = `${targetRect.bottom - 10 - 2}px`;
-                }
 
-                $flyout.addClass('bottom-up');
-            } else {
-                $flyout.removeClass('bottom-up');
-                css.top = targetRect.top;
+    /**
+     * Show the fly-out menu for the given target navigation item
+     *
+     * @param $target
+     */
+    Navigation.prototype.showFlyoutMenu = function($target) {
+        const $flyout = $target.find('.nav-level-2');
+        if (! $target.is(':hover')) {
+            return;
+        }
+
+        $target.siblings().not($target).removeClass('hover');
+        $target.addClass('hover');
```

Comment:
I've meant to move multiple lines before. Take a look at the noted range of lines in the screenshot. It's obviously bugged right now, since the diff is outdated, but it's clearly not just a single line what this comment is about!

![image](https://github.com/user-attachments/assets/c6ef123e-56b8-40e2-b147-a4247be8439f)

Now take note of the range of **this** comment: +376 to +381

Reply:
I see, but I think the removal of the hover class from the sibling navigation item and adding that class to current target needs to be done in `showFlyoutMenu` method. Orelse I have to do that in onMouseMove method again. That is why I misunderstood the comment. But the lines from +376 to +379 could be moved to onMouseEnter.